system/nxplayer/nxplayer.c

```diff
@@ -221,18 +220,35 @@ static int _open_with_http(const char *fullurl)
 
   FAR struct hostent *he;
   he = gethostbyname(hostname);
+  if (!he)
+    {
+      close(s);
+      return -ENETUNREACH; /* DNS resolution failed */
+    }
 
   memcpy(&server.sin_addr.s_addr,
          he->h_addr, sizeof(in_addr_t));
-
   n = connect(s,
               (struct sockaddr *)&server,
               sizeof(struct sockaddr_in));
-
   if (-1 == n)
     {
+      int err = errno;
       close(s);
-      return -1;
+      switch (err)
```

Comment:
Please take a look what switch format is preferred https://nuttx.apache.org/docs/latest/contributing/coding_style.html#switch-statement